rust/diesel/src/order.rs

```diff
@@ -0,0 +1,173 @@
+use bigdecimal::BigDecimal;
+use diesel;
+use diesel::pg::PgConnection;
+use diesel::prelude::*;
+use serde::{Deserialize, Serialize};
+use std::ops::{Add, Mul};
+
+use crate::product::Product;
+use crate::schema::order_lines;
+use crate::schema::orders;
+use num_traits::identities::Zero;
+
+#[derive(AsChangeset, Serialize, Deserialize, Queryable, Insertable)]
+#[serde(rename_all = "camelCase")]
+pub struct Order {
+    pub order_id: i32,
+    pub order_total: BigDecimal,
+    pub user_id: i32,
+}
+
+#[table_name = "orders"]
+#[derive(AsChangeset, Serialize, Deserialize, Queryable, Insertable)]
+#[serde(rename_all = "camelCase")]
+struct NewOrder {
+    pub order_total: BigDecimal,
+    pub user_id: i32,
+}
+
+#[table_name = "order_lines"]
+#[derive(AsChangeset, Serialize, Deserialize, Queryable, Insertable)]
+#[serde(rename_all = "camelCase")]
+pub struct NewOrderLine {
+    pub order_id: i32,
+    pub product_id: i32,
+    pub units: i16,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NewUserOrder {
+    pub user_id: i32,
+    pub products: Vec<ProductOrder>,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ProductOrder {
+    pub product_id: i32,
+    pub units: i16,
+}
+
+impl Order {
+    pub fn create(order: NewUserOrder, connection: &PgConnection) -> Option<NewUserOrder> {
```

Comment:
I think we should return a `Result<Order, ...>` rather than a `Option<NewUserOrder>`. For one, if an error occurs and the order can't be created, we probably want to know why. For two, if we are calling `Order::create(...)` it is confusing that the result we get back is not of type `Order`.

Reply:
Again, in this example, it is sufficient to know if the operation failed, not why it failed, hence I will keep Option here

Reply:
Do we not expect this to ever fail? When I left this (and the other related) comments I was thinking of a case where a user clones this code and follows the readme and then (for whatever reason) see's something along the lines of "panic: attempted to unwrap None". I'm not sure we could expect someone testing the app to go into the code and add logging or better error handling in order to properly debug their error.

Reply:
For reference, I was concerned about this because I encountered a similarly unhelpful error when I tried to install and run this with a yugabyte database. Once you have this code working with yugabyte, however, if you think errors like these should not occur, then then I think it is OK to ignore/hide them.

Reply:
This `unwrap()` was inside of a transaction, and it is replaced with `?` in next iteration

Reply:
Talked offline with @rkarthik007 and we are going to let error reporting/propagating slide for now in the interest of time, but I will open an issue for it to be done in the future.